### PR TITLE
NIFI-10371: When a component is moved between groups, ensure that its…

### PR DESCRIPTION
**Pull request by markap14**
… versioned component id is unique within the destination group. Also ensure that when adding a connection to a PG with the VersionedComponentSynchronizer that we prefer obtaining source/destination by instance id instead of versioned id.

# Summary

[NIFI-00000](https://issues.apache.org/jira/browse/NIFI-00000)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [ ] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [ ] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [ ] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [ ] Pull Request based on current revision of the `main` branch
- [ ] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [ ] Build completed using `mvn clean install -P contrib-check`
  - [ ] JDK 8
  - [ ] JDK 11
  - [ ] JDK 17

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

**Comment by joewitt** [comment]
full clean build with contrib check and full suite of system test integration tests pass on local machine.  Assuming all checks pass i plan to merge it.  Code changes make sense as doc'd and the test seems to cover the affected case.

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/groups/StandardProcessGroup.java [hunk 1]
```diff
@@ -971,6 +978,50 @@ public void addProcessor(final ProcessorNode processor) {
...
+        if (!optionalVersionControlId.isPresent()) {
+            return;
+        }
+
+        final String versionControlId = optionalVersionControlId.get();
+        final boolean duplicateId = containsVersionedComponentId(componentsToCheck, versionControlId);
+
+        if (duplicateId) {
+            LOG.debug("Adding {} to {}, found conflicting Version Component ID {} so marking Version Component ID of {} as null", component, this, versionControlId, component);
+            setVersionedComponentId(null);
```
Comment (exceptionfactory) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/groups/StandardProcessGroup.java:1007]: This line sets the `versionedComponentId` of the ProcessGroup to `null`, as opposed to setting the `versionedComponentId` of the `VersionedComponent` to `null`, which causes the new test in `StandardProcessGroupIT` to fail.

**Review by exceptionfactory — CHANGES_REQUESTED** [review]
Thanks for addressing this issue @markap14! Most of the changes look good, but running `StandardProcessGroupIT` resulted in a failure as noted in `ensureUniqueVersionControlId`.

**Review by exceptionfactory — APPROVED** [review]
Thanks for making the adjustment @markap14, the latest version looks good! +1 merging

**Pull request closed without merging**